examples/src/main/scala/com/cloudera/livy/apps/WordCountApp.scala

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to Cloudera, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Cloudera, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cloudera.livy.apps
+
+import java.io.{File, FileNotFoundException}
+import java.net.URI
+import java.security.CodeSource
+import java.util.logging.Logger
+
+import org.apache.spark.storage.StorageLevel
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+import com.cloudera.livy.LivyClientBuilder
+import com.cloudera.livy.scalaapi.{LivyScalaClient, ScalaJobHandle, _}
+
+/**
+  * A WordCount example using Scala-API which reads text from a stream and saves
+  * it as data frames. The word with maximum count is the result
+  */
+object WordCountApp {
+
+  var scalaClient: LivyScalaClient = null
+  val logger = Logger.getLogger("WordCountApp")
+
+  /**
+    * Initializes the scala client with the given url
+    * @param url The livy server url
+    */
+  def init(url: String): Unit = {
+    scalaClient = new LivyClientBuilder(true).setURI(new URI(url)).build().asScalaClient
+  }
+
+  /**
+    * Uploads the scala-api Jar and the examples Jar from the target directory
+    * @throws FileNotFoundException If either of scala-api Jar or examples Jar is not found
+    */
+  @throws(classOf[FileNotFoundException])
+  def uploadRelevantJarsForJobExecution(): Unit = {
+    val exampleAppJarPath = getPath(this)
+    val scalaApiJarPath =  getPath(scalaClient)
+    uploadJar(exampleAppJarPath)
+    uploadJar(scalaApiJarPath)
+  }
+
+  @throws(classOf[FileNotFoundException])
+  private def getSourceIfNotNull(obj: Object): CodeSource = {
```

Comment:
nit: `ifNotNull` in the method name is redundant given that the method throws an exception.